File: DynamicKey/AgoraDynamicKey/nodejs/src/RtmTokenBuilder.js
const AccessToken = require("../src/AccessToken").AccessToken
const Priviledges = require('../src/AccessToken').priviledges

const Role = {
  Rtm_User: 1
}
class RtmTokenBuilder {

  /**
   * @param {*} appID: The App ID issued to you by Agora. Apply for a new App ID from 
   *       Agora Dashboard if it is missing from your kit. See Get an App ID.
   * @param {*} appCertificate:	Certificate of the application that you registered in 
   *                 the Agora Dashboard. See Get an App Certificate.
   * @param {*} account: The user account. 
   * @param {*} privilegeExpiredTs : represented by the number of seconds elapsed since 
   *                   1/1/1970. If, for example, you want to access the
   *                   Agora Service within 10 minutes after the token is 
   *                   generated, set expireTimestamp as the current 
   * @return token
   */
  static buildToken(appID, appCertificate, account, privilegeExpiredTs) {
    const key = new AccessToken(appID, appCertificate, '', account)
    key.addPriviledge(Priviledges.kRtmLogin, privilegeExpiredTs)
    return key.build()
  }
}

module.exports.RtmTokenBuilder = RtmTokenBuilder
module.exports.Role = Role

File: DynamicKey/AgoraDynamicKey/dart/lib/src/rtm_token_builder.dart
import 'access_token.dart';

/// Builds Token007 tokens for RTM services.
class RtmTokenBuilder {
  /// Builds an RTM token for a user account.
  ///
  /// [appId] is the App ID issued by Agora; create one in Agora Dashboard if needed.
  /// [appCertificate] is the certificate of the application registered in Agora Dashboard.
  /// [userId] is the user's account and must not exceed 64 bytes.
  /// [tokenExpireSeconds] is the expiration time in seconds from now; use `600` for 10 minutes.
  ///
  /// Returns the generated RTM token.
  static String buildToken({
    required String appId,
    required String appCertificate,
    required String userId,
    required int tokenExpireSeconds,
  }) {
    final token = AccessToken.create(
      appId,
      appCertificate,
      expire: tokenExpireSeconds,
    );
    final service = ServiceRtm(userId)
      ..addPrivilege(ServiceRtm.privilegeLogin, tokenExpireSeconds);
    token.addService(service);

    return token.build();
  }

  /// Builds an RTM2 token with resource-level permissions.
  ///
  /// This special interface requires Agora assistance for proper usage.
  /// [appId] is the App ID issued by Agora.
  /// [appCertificate] is the application certificate registered in Agora Dashboard.
  /// [userId] is the user's account and must not exceed 64 bytes.
  /// [permissions] contains the RTM2 resource-level permissions.
  /// [tokenExpireSeconds] is the expiration time in seconds from now.
  static String buildTokenWithPermissions({
    required String appId,
    required String appCertificate,
    required String userId,
    required Rtm2Permissions permissions,
    required int tokenExpireSeconds,
  }) {
    final token = AccessToken.create(
      appId,
      appCertificate,
      expire: tokenExpireSeconds,
    );
    final service = ServiceRtm2(userId, permissions)
      ..addPrivilege(ServiceRtm2.privilegeLogin, tokenExpireSeconds);
    token.addService(service);
    return token.build();
  }
}
